lib/strerror.c

```diff
@@ -453,6 +453,78 @@ curl_share_strerror(CURLSHcode error)
 #endif
 }
 
+const char *
+curl_url_strerror(CURLUcode error)
+{
+#ifndef CURL_DISABLE_VERBOSE_STRINGS
+  switch(error) {
+  case CURLUE_OK:
+    return "No error";
+
+  case CURLUE_BAD_HANDLE:
+    return "An argument that should be a CURLU pointer was passed in as a NULL";
+
+  case CURLUE_BAD_PARTPOINTER:
+    return "A NULL pointer was passed to the 'part' argument of curl_url_get";
```

Comment:
I propose:

"An invalid part pointer was given as argument"

This way this can also be for other errors than NULL.

Reply:
how about combine them into CURLU_BAD_FUNCTION_ARGUMENT

Reply:
At this point that would rather add a third error code so I'm not sure that helps anything...

Reply:
Ah, I thought those two were being added.

Reply:
I've synced text with #7611, but it reads slightly different to @bagder's proposal here. I hope it's fine.